common/djangoapps/microsite_configuration/tests/backends/test_base.py

```diff
@@ -130,3 +138,36 @@ def test_not_yet_implemented(self):
 
         with self.assertRaises(NotImplementedError):
             backend.get_all_orgs()
+
+
+@patch(
+    'microsite_configuration.microsite.BACKEND',
+    microsite.get_backend(
+        'microsite_configuration.backends.base.BaseMicrositeBackend', BaseMicrositeBackend
+    )
+)
+class BaseMicrositeBackendTests(TestCase):
+    """
+    Go through and test the BaseMicrositeBackend class for behavior which is not
+    overriden in subclasses
+    """
+    def test_enable_microsites_pre_startup(self):
+        """
+        Tests microsite.test_enable_microsites_pre_startup works as expected.
+        """
+        # remove microsite root directory paths first
+        settings.DEFAULT_TEMPLATE_ENGINE['DIRS'] = [
+            path for path in settings.DEFAULT_TEMPLATE_ENGINE['DIRS']
+            if path != settings.MICROSITE_ROOT_DIR
+        ]
+
+        with patch('microsite_configuration.backends.base.BaseMicrositeBackend.has_configuration_set',
```

Comment:
@felipemontoya my concern here is that we are only testing the situation when `has_configuration_set` returns `True`  or `False`. However, I think it would a good idea to test the following user cases,
1. User sets/leaves-empty the value of `settings.MICROSITE_CONFIGURATION ` 
2. User populates/leaves-empty configuration in `Microsite` table.

here we can patch `settings.MICROSITE_CONFIGURATION ` and in the tests for `microsite_configuration.backends.database.DatabaseMicrositeBackend` we can add/remove configuration in `Microsite` table. 
Maybe we can leave the above test case as it is and add new ones for above user cases.